apps/docs/src/app/platform/component-docs/platform-split-menu-button/platform-split-menu-button-examples/platform-split-menu-button-examples.component.html

```diff
@@ -1,98 +1,197 @@
+    <fdp-menu-item (itemSelect)="onItemSelectCompactStandard('Standard Button 2')">Standard Button 2</fdp-menu-item>
+    <fdp-menu-item (itemSelect)="onItemSelectCompactStandard('Standard Button 3')">Standard Button 3</fdp-menu-item>
 </fdp-menu>
 
-<fdp-menu #basicMenuCompact id="basic-menu">
-    <fdp-menu-item (itemSelect)="onItemSelectCompact('First Item')">First Item</fdp-menu-item>
-    <fdp-menu-item (itemSelect)="onItemSelectCompact('Second Item')">Second Item</fdp-menu-item>
-    <fdp-menu-item (itemSelect)="onItemSelectCompact('Third Item')">Third Item</fdp-menu-item>
+<fdp-menu #basicMenuCompactPositive id="basic-menu-Compact-positive" xPosition="before">
+    <fdp-menu-item (itemSelect)="onItemSelectCompactPositive('Positive Button')">Positive Button</fdp-menu-item>
+    <fdp-menu-item (itemSelect)="onItemSelectCompactPositive('Positive Button 2')">Positive Button 2</fdp-menu-item>
+    <fdp-menu-item (itemSelect)="onItemSelectCompactPositive('Positive Button 3')">Positive Button 3</fdp-menu-item>
+</fdp-menu>
+<fdp-menu #basicMenuCompactNegative id="basic-menu-Compact-negative" xPosition="before">
+    <fdp-menu-item (itemSelect)="onItemSelectCompactNegative('Negative Button')">Negative Button</fdp-menu-item>
+    <fdp-menu-item (itemSelect)="onItemSelectCompactNegative('Negative Button 2')">Negative Button 2</fdp-menu-item>
+    <fdp-menu-item (itemSelect)="onItemSelectCompactNegative('Negative Button 3')">Negative Button 3</fdp-menu-item>
+</fdp-menu>
+<fdp-menu #basicMenuCompactGhost id="basic-menu-Compact-ghost" xPosition="before">
+    <fdp-menu-item (itemSelect)="onItemSelectCompactGhost('Ghost Button')">Ghost Button</fdp-menu-item>
+    <fdp-menu-item (itemSelect)="onItemSelectCompactGhost('Ghost Button 2')">Ghost Button 2</fdp-menu-item>
+    <fdp-menu-item (itemSelect)="onItemSelectCompactGhost('Ghost Button 3')">Ghost Button 3</fdp-menu-item>
+</fdp-menu>
+<fdp-menu #basicMenuCompactTransparent id="basic-menu-Compact-transparent" xPosition="before">
+    <fdp-menu-item (itemSelect)="onItemSelectCompactTransparent('Transparent Button')">Transparent Button
+    </fdp-menu-item>
+    <fdp-menu-item (itemSelect)="onItemSelectCompactTransparent('Transparent Button 2')">Transparent Button 2
+    </fdp-menu-item>
+    <fdp-menu-item (itemSelect)="onItemSelectCompactTransparent('Transparent Button 3')">Transparent Button 3
+    </fdp-menu-item>
+</fdp-menu>
+<fdp-menu #basicMenuCompactEmphasized id="basic-menu-Compact-emphasized" xPosition="before">
+    <fdp-menu-item (itemSelect)="onItemSelectCompactEmphasized('Emphasized Button')">Emphasized Button</fdp-menu-item>
+    <fdp-menu-item (itemSelect)="onItemSelectCompactEmphasized('Emphasized Button 2')">Emphasized Button 2
+    </fdp-menu-item>
+    <fdp-menu-item (itemSelect)="onItemSelectCompactEmphasized('Emphasized Button 3')">Emphasized Button 3
+    </fdp-menu-item>
 </fdp-menu>
```

Comment:
I believe we should avoid examples with 200 lines of code, that's not readable at all, I just got lost there in one minute :)

If we want to show available `button types` we should show just it. We shouldn't repeat it twice for `comact` mode if it was already shown in `cozy` mode cause it does not bring any additional value to user.

Could we reuse one `fdp-menu` for all buttons to reduce the example code.

Reply:
Hi dima,
i can split the file to have examples of cozy and compact different.
but for meeting testing team requirement, i have to include examples for all states and for cozy-compact both.

for having a single "fdp-menu":
in current implementation i am calling method from menu-item selection, and this method updates the label for primary button.
if i keep a single fdp-menu, then selecting a item in one split-menu-button, will change label for all other split-menu-button as well. which will be wrong. so i have to keep separate fdp-menu.

I am raising a separate PR, where feature for updating the label for primary button will be moved inside component.
As this PR was part of defect hunting, i will not be doing this change in this PR.

Reply:
I see your point, sounds good

Reply:
@DeepakSap14 

> but for meeting testing team requirement, i have to include examples for all states and for cozy-compact both.

If create these examples for end users (app developers) we should not adapt it for our testing team.
I believe testing team should have additional tools to test it properly.